### DISCUSSION TITLE
How is add_lora_to_model() called?

### DISCUSSION THREAD

**Discussion by joesocktwo** [oobabooga/text-generation-webui#4646] [Q&A]
I'm trying to figure out exactly how the LoRAs get applied to the model in the code base... My goal is to be able to trigger LoRA loads through the API, but I'm not really sure where to start...

Any clues?

**Comment by FartyPants** [comment]
You can look at my 
https://github.com/FartyPants/VirtualLora
and follow the def Load_and_apply_lora()
I don't know how the main does it now as there had been some changes in PR, but this is the correct way and it has a lot of error checking and print so you can see what it does step by step.
